packages/frontend/web/src/components/profile-page/modal/FormPart.tsx

```diff
@@ -0,0 +1,114 @@
+import type { FieldErrors, UseFormRegister } from 'react-hook-form';
+
+import type { UserInfoForm } from '@app/types';
+
+interface FormPartPros {
+  readonly register: UseFormRegister<UserInfoForm>;
+  readonly errors: FieldErrors<UserInfoForm>;
+}
+
+export default function FormPart({ register, errors }: FormPartPros) {
+  const inputs = [
+    {
+      label: 'email',
+      type: 'email',
+      name: 'email',
+      errors: [],
+      errorsMessage: '',
+    },
+    {
+      label: 'pseudo',
+      type: 'text',
+      name: 'pseudo',
+      errors: [
+        errors.pseudo?.type === 'maxLength',
+        errors.pseudo?.type === 'minLength',
+      ],
+      errorsMessage: errors.pseudo?.message,
+    },
+    {
+      label: 'actual password',
+      type: 'password',
+      name: 'actualPassword',
+      errors: [
+        errors.actualPassword?.type === 'maxLength',
+        errors.actualPassword?.type === 'minLength',
+        errors.actualPassword?.type === 'validate',
+      ],
+      errorsMessage: errors.actualPassword?.message,
+    },
+    {
+      label: 'new password',
+      type: 'password',
+      name: 'newPassword',
+      errors: [
+        errors.newPassword?.type === 'maxLength',
+        errors.newPassword?.type === 'minLength',
+        errors.newPassword?.type === 'validate',
+      ],
+      errorsMessage: errors.newPassword?.message,
+    },
+    {
+      label: 'confirm new password',
+      type: 'password',
+      name: 'confirmNewPassword',
+      errors: [
+        errors.confirmNewPassword?.type === 'maxLength',
+        errors.confirmNewPassword?.type === 'minLength',
+      ],
+      errorsMessage: errors.confirmNewPassword?.message,
+    },
+  ];
+
+  return (
+    <div className='w-[80%] flex-col items-center gap-3 lg:flex'>
+      {inputs.map((input) => (
+        <div key={input.label}>
+          <div className='w-full items-center justify-between gap-3 lg:flex'>
+            <label
+              className={`font-stroke 2xl:text-md ${
+                input.label === 'email' || input.label === 'pseudo'
+                  ? 'mr-10'
+                  : 'false'
+              } w-[100px] text-sm uppercase text-white md:text-xs lg:mr-0 lg:text-sm`}
+            >
+              {input.label}
+            </label>
+            <div className='flex flex-col'>
+              <input
+                type={input.type}
+                readOnly={input.type === 'email' ? true : false}
+                className='border-dark w-full rounded-sm  border-[4px] p-1 md:w-[80%]'
+                {...register(
+                  input.name as
+                    | 'pseudo'
+                    | 'email'
+                    | 'actualPassword'
+                    | 'newPassword'
+                    | 'confirmNewPassword',
```

Comment:
```suggestion
                  input.name as keyof UserInfoForm
```
Un truc du genre devrait fonctionner